File: backend/Models/Transaction.cs
namespace Expense.Api.Models
{
    public enum TransactionType
    {
        Income,
        Expense
    }

    // Gasto ou receita ligado a uma pessoa.
    public class Transaction
    {
        public int Id { get; set; }
        public string Description { get; set; } = string.Empty;
        public decimal Amount { get; set; }
        public TransactionType Type { get; set; }

        public int PersonId { get; set; }
        public Person? Person { get; set; }
    }
}

File: frontend/src/types/Transaction.ts
// Enum para o tipo de transação
export enum TransactionType {
    Income = 0,
    Expense = 1,
}

// Interface para a transação
export interface Transaction {
    id: number;
    description: string;
    amount: number;
    type: TransactionType;
    personId: number;
    personName: string;
}

export interface CreateTransactionDto {
    description: string;
    amount: number;
    type: TransactionType;
    personId: number; 
}